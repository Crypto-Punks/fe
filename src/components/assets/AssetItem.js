import React from 'react';
import PropTypes from 'prop-types';

const AssetItem = ({ logo, name, amount, value }) => {
  return (
    <li>
      <img src={logo} alt={name} />
      <p>{amount} {name}</p>
      <p>{value}</p>
    </li>
  );
};

AssetItem.propTypes = {
  logo: PropTypes.string,
  name: PropTypes.string.isRequired,
  amount: PropTypes.number.isRequired,
  value: PropTypes.string.isRequired
};

export default AssetItem;
