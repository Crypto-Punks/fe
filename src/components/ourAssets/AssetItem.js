import React from 'react';
import PropTypes from 'prop-types';

const AssetItem = ({ activeCoin }) => {
  const { logo, name, amount, value } = activeCoin;
  return (
    <li>
      <img src={logo} alt={name} />
      <p>{amount} {name}</p>
      <p>{value}</p>
    </li>
  );
};

AssetItem.propTypes = {
  activeCoin: PropTypes.shape({
    logo: PropTypes.string,
    name: PropTypes.string.isRequired,
    amount: PropTypes.number.isRequired,
    value: PropTypes.string.isRequired
  })
};

export default AssetItem;
