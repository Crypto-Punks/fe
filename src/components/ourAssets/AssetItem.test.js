import React from 'react';
import { shallow } from 'enzyme';
import AssetItem from './AssetItem';

const data = {
  activeCoin: {}
};

describe('AssetItem component', () => {
  it('renders AssetItem', () => {
    const wrapper = shallow(<AssetItem { ...data} />);
    expect(wrapper).toMatchSnapshot();
  });
});
