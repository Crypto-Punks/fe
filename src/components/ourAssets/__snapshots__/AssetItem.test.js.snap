// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetItem component renders AssetItem 1`] = `
<Link
  to="detail/undefined"
>
  <li>
    <img />
    <p
      className="Name"
    >
       (
      )
    </p>
    <p
      className="Change"
    >
      $
      NaN
    </p>
  </li>
</Link>
`;
